src/LinuxTracing/LinuxTracingUtilsTest.cpp

```diff
@@ -56,14 +56,16 @@ TEST(GetThreadState, LinuxTracingTestsMainAndAnother) {
     // Make sure /proc/<pid>/stat is parsed correctly
     // even when the thread name contains spaces and parentheses.
     pthread_setname_np(pthread_self(), ") )  )()( )(  )");
+    // Give the main thread the time to get to the mutex.
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
     {
       absl::MutexLock lock{&mutex};
       thread_tid = syscall(SYS_gettid);
       thread_state_holding_mutex = GetThreadState(thread_tid);
       main_state_waiting_mutex = GetThreadState(main_tid);
     }
     // Give the main thread the time to read this thread's state before exiting.
-    std::this_thread::sleep_for(std::chrono::milliseconds(50));
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
Could the same be achieved by locking a second mutex that was previosuly locked on the main thread? I guess that would also put the thread into a sleep state?

Reply:
Yes, but the idea here was really to get the state while sleeping, the state while holding a mutex is already tested with `main_state_waiting_mutex`.

Reply:
Ahh. Sorry. I didn't get that. Please ignore my comment. :see_no_evil:

Reply:
Because it's not written anywhere. I'll add a comment.

Reply:
Done.